lib/utils/error_handler.ts

```diff
@@ -5,10 +5,29 @@ import * as express from "express";
  *
  * @param response The Express Response for the current request
  */
-export function handleErrorAndRespond(response: express.Response): (reason: any) => any {
-  return (reason: any) => {
-    response.status(500).json({
+export function handleErrorAndRespond<T>(response: express.Response): (reason: T) => express.Response {
```

Comment:
probably it's better to have an [express error handler](http://expressjs.com/en/guide/error-handling.htm) so you don't have to call it explicitly, you just `throw new ApiError(status, reason)` inside the controller.